samza-core/src/main/scala/org/apache/samza/config/MetricsConfig.scala

```diff
@@ -31,6 +31,7 @@ object MetricsConfig {
   val METRICS_SNAPSHOT_REPORTER_INTERVAL= "metrics.reporter.%s.interval"
   val METRICS_TIMER_ENABLED= "metrics.timer.enabled"
   val METRICS_SNAPSHOT_REPORTER_BLACKLIST = "metrics.reporter.%s.blacklist"
+  val METRICS_SNAPSHOT_REPORTER_NAME_FOR_DIAGNOSTICS = "diagnosticsreporter"
```

Comment:
Maybe "metrics.diagnostics.reporter"? Or if you don't consider this as "metrics", then it might be better to put the constant somewhere else.